### PR TITLE
[RW-7268][risk=no] resolve e2e test failures

### PR DESCRIPTION
**Pull request by aweng98**

### CODE REVIEW

File: e2e/app/page/workspace-data-page.ts [hunk 1]
```diff
@@ -33,8 +33,8 @@ export default class WorkspaceDataPage extends WorkspaceBase {
 
   async imgDiagramLoaded(): Promise<ElementHandle[]> {
     return Promise.all<ElementHandle, ElementHandle>([
-      this.page.waitForXPath('//img[@src="/assets/images/dataset-diagram.svg"]', { visible: true }),
-      this.page.waitForXPath('//img[@src="/assets/images/cohort-diagram.svg"]', { visible: true })
+      this.page.waitForXPath('//img[@data-test-id="dataset-diagram"]', { visible: true }),
+      this.page.waitForXPath('//img[@data-test-id="cohort-diagram"]', { visible: true })
```
Comment (aweng98) [inline e2e/app/page/workspace-data-page.ts:37]: Update xpath to the latest UI.

File: e2e/utils/waits-utils.ts [hunk 2]
```diff
@@ -380,7 +381,7 @@ export async function waitWhileLoading(
   const { waitForRuntime = false } = opts;
   const notBlankPageSelector = '[data-test-id="sign-in-container"], title:not(empty), div.spinner, svg[viewBox]';
   const spinElementsSelector = `[style*="running spin"], .spinner:empty, [style*="running rotation"]${
-    waitForRuntime ? '' : ':not([aria-hidden="true"]):not([data-test-id="runtime-status-icon"])'
+    waitForRuntime ? '' : ':not([aria-hidden="true"]):not([data-test-id="runtime-status-icon-container"])'
```
Comment (aweng98) [inline e2e/utils/waits-utils.ts:384]: Update xpath to the latest UI.

File: e2e/utils/waits-utils.ts [hunk 3]
```diff
@@ -392,29 +393,31 @@ export async function waitWhileLoading(
...
+    logger.error(`FAILED to wait for spinner stop: xpath="${spinElementsSelector}"`);
     if (err.message.includes('Target closed')) {
       // Leave blank. Ignore error and continue test.
       // Puppeteer can throw following exception when polling for mutation status if this object disappeared in DOM
       //   or page navigation happened.
       // Error: Protocol error (Runtime.callFunctionOn): Target closed.
     } else {
       logger.error(err.stack);
-      throw err;
+      await takeScreenshot(page, 'Spinner_TimeoutError.jpg');
```
Comment (aweng98) [inline e2e/utils/waits-utils.ts:417]: Help with failure troubleshooting.

File: e2e/utils/waits-utils.ts [hunk 4]
```diff
@@ -392,29 +393,31 @@ export async function waitWhileLoading(
...
-      (css) => {
-        return !document.querySelectorAll(css).length;
-      },
-      { polling: 'mutation', timeout },
-      spinElementsSelector
-    ),
-    page.waitForSelector(spinElementsSelector, { hidden: true, timeout })
-  ]).catch((err: Error) => {
-    logger.error(`Failed wait for spinner stop: xpath="${spinElementsSelector}"`);
+  try {
```
Comment (aweng98) [inline e2e/utils/waits-utils.ts:396]: Update to use `try/catch` instead promise `.catch` to catch asyn and non-asyn errors.

**Review by als364 — APPROVED** [review]

**Pull request merged**